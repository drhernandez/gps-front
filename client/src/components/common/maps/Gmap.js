/*global google*/
import React from "react";
import { compose, withProps } from "recompose"
import { 
  withScriptjs,
  withGoogleMap, 
  GoogleMap, 
  Marker 
} from "react-google-maps"
import HeatmapLayer from "react-google-maps/lib/components/visualization/HeatmapLayer";

const mapKey = process.env.REACT_APP_GOOGLE_MAP_KEY

const Gmap = compose(
  withProps({
    googleMapURL: `https://maps.google.com/maps/api/js?libraries=visualization&key=${mapKey}`,
    loadingElement: <div style={{ height: `100%` }} />,
    containerElement: <div style={{ height: `450px` }} />,
    mapElement: <div style={{ height: `100%` }} />,
  }),
  withScriptjs,
  withGoogleMap
)((props) =>
  <GoogleMap
    key='map-component'
    zoom={props.zoom || 12}
    center={props.center || { lat: -31.412130, lng: -64.188510 }}
  >
    {props.isMarkerShown && props.markers.map((marker) => (
      <Marker key={marker.id} position={marker.position} icon={marker.icon} animation={marker.animation} />
    ))}

    {props.isHeatMapLayerShown &&
      <HeatmapLayer data={props.trackings.map((tracking) => (
        new google.maps.LatLng(tracking.lat, tracking.lng)
      ))} />
    }
  </GoogleMap>
);

export default Gmap;

