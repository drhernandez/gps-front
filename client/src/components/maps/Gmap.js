/*global google*/
import React from "react";
import { 
  withGoogleMap, 
  GoogleMap, 
  Marker 
} from "react-google-maps"
import HeatmapLayer from "react-google-maps/lib/components/visualization/HeatmapLayer";

const Gmap = withGoogleMap((props) =>
  <GoogleMap
    zoom={props.zoom || 12}
    center={props.center || { lat: -31.412130, lng: -64.188510 }}
  >
    {props.isMarkerShown && props.markers.map((marker) => (
      <Marker key={marker.id} position={marker.position} icon={marker.icon} animation={marker.animation} />
    ))}

    {props.isHeatMapLayerShown &&
      <HeatmapLayer data={props.trackings.map((tracking) => (
        new google.maps.LatLng(tracking.lat, tracking.lng)
      ))} />
    }
  </GoogleMap>
);

export default Gmap;