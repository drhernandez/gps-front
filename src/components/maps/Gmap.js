import React from "react";
import {
  Card, 
  CardHeader,
  CardBody,
  FormSelect,
  Form,
} from "shards-react";

const GoogleMap = (props) => (
  <div>
    <Card small className="mb-4">
      <CardHeader>
        <Form className="add-new-post">
          <FormSelect id="feInputState">
            <option value="null">Elija un veh√≠culo...</option>
            <option value="ford">Ford</option>
            <option value="toyota">Toyota</option>
          </FormSelect>
        </Form>
      </CardHeader>
      <CardBody>
        Hola
      </CardBody>
    </Card>
  </div>
);

export default GoogleMap;