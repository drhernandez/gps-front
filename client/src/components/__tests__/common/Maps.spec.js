import React from 'react'
import renderer from 'react-test-renderer'

import Gmap from '../../common/maps/Gmap'

describe('Maps components', () => {

  it('Gmap snapshot', async () => {
    const componet = renderer.create(
      <Gmap />
    )

    expect(componet.toJSON).toMatchSnapshot()
  })
})