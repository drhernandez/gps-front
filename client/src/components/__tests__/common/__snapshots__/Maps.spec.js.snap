// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Maps components Gmap snapshot 1`] = `[Function]`;
